x-pack/plugins/task_manager/server/lib/intervals.test.ts

```diff
@@ -65,6 +66,44 @@ describe('taskIntervals', () => {
     });
   });
 
+  describe('parseIntervalAsMinute', () => {
+    test('it accepts intervals in the form `Nm`', () => {
+      expect(() => parseIntervalAsMinute(`${_.random(1, 1000)}m`)).not.toThrow();
+    });
+
+    test('it accepts intervals in the form `Ns`', () => {
+      expect(() => parseIntervalAsMinute(`${_.random(1, 1000)}s`)).not.toThrow();
+    });
+
+    test('it rejects 0 based intervals', () => {
+      expect(() => parseIntervalAsMinute('0m')).toThrow(
+        /Invalid interval "0m"\. Intervals must be of the form {number}m. Example: 5m/
+      );
+      expect(() => parseIntervalAsMinute('0s')).toThrow(
+        /Invalid interval "0s"\. Intervals must be of the form {number}m. Example: 5m/
+      );
+    });
+
+    test('it rejects intervals are not of the form `Nm` or `Ns`', () => {
+      expect(() => parseIntervalAsMinute(`5m 2s`)).toThrow(
+        /Invalid interval "5m 2s"\. Intervals must be of the form {number}m. Example: 5m/
+      );
+      expect(() => parseIntervalAsMinute(`hello`)).toThrow(
+        /Invalid interval "hello"\. Intervals must be of the form {number}m. Example: 5m/
+      );
+    });
+
+    test('returns an interval as m', () => {
+      expect(parseIntervalAsMinute('5s')).toEqual(5 / 60);
+      expect(parseIntervalAsMinute('15s')).toEqual(15 / 60);
+      expect(parseIntervalAsMinute('20m')).toEqual(20);
+      expect(parseIntervalAsMinute('61m')).toEqual(61);
+      expect(parseIntervalAsMinute('90m')).toEqual(90);
+      expect(parseIntervalAsMinute('2h')).toEqual(2 * 60);
+      expect(parseIntervalAsMinute('9d')).toEqual(9 * 60 * 24);
+    });
```

Comment:
any need to protect against / validate negative numbers?  
what about '2.5m'?
or if just 'm' is passed in?
And special chars?  
I won't drone on (yet) I don't know where / how the payloads for the api call are created so won't go overboard.  can you link me? I'm lost in the 20 files.  :)

Reply:
There is validation for the intervals that are passed in. It will throw an error if you try to pass in a negative number, a decimal, or just 'm'. It expects intervals in the format `{number}{m, s, h, or d}`. All of this happens here: `x-pack/plugins/task_manager/server/lib/intervals.ts`. I am using the minute interval parsing here: `x-pack/plugins/task_manager/server/monitoring/background_task_utilization_statistics.ts`, where we parse the task interval to determine the number of tasks/minute